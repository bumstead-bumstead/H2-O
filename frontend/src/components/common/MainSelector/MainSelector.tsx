import type { HTMLAttributes, PropsWithChildren } from 'react';
import styled from '@emotion/styled';

interface Props extends HTMLAttributes<HTMLDivElement> {
  title?: string;
}

function MainSelector({ children, title, ...restProps }: PropsWithChildren<Props>) {
  return (
    <MainSelectorContainer {...restProps}>
      <div>
        {title && <h2 className='title'>{title}</h2>}
        {children}
      </div>
    </MainSelectorContainer>
  );
}

export default MainSelector;

const MainSelectorContainer = styled.div`
  position: relative;
  width: 100%;
  min-height: calc(100vh - 420px);
  padding-top: 16px;
  background-color: ${({ theme }) => theme.colors.white};
  filter: drop-shadow(0px 0px 8px rgba(131, 133, 136, 0.2));

  & > div {
    max-width: 1024px;
    width: 100%;
    margin: 0 auto;
  }

  .title {
    ${({ theme }) => theme.typography.HeadKRMedium16}
    color: ${({ theme }) => theme.colors.gray900};
    padding-bottom: 12px;
  }
`;
