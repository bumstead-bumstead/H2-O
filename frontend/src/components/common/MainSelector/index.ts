export { default as MainSelector } from './MainSelector';
