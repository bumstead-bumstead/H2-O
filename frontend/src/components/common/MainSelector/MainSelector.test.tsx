import MainSelector from './MainSelector';
import { render, screen } from '@/tests/test-util';

describe('MainSelector 컴포넌트 테스트', () => {
  it('MainSelector가 정상적으로 렌더링된다.', () => {
    render(<MainSelector data-testid='main' />);

    expect(screen.getByTestId('main')).toBeInTheDocument();
  });

  it('title props를 전달하면 타이틀 텍스트가 정상적으로 렌더링된다.', () => {
    render(<MainSelector data-testid='main' title='타이틀' />);

    expect(screen.getByText('타이틀')).toBeInTheDocument();
  });

  it('MainSelector 스냅샷 테스트', () => {
    render(<MainSelector data-testid='main' />);

    expect(screen.getByTestId('main')).toMatchSnapshot();
  });
});
