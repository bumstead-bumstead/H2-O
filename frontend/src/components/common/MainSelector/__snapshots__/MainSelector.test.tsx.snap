// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainSelector 컴포넌트 테스트 > MainSelector 스냅샷 테스트 1`] = `
.emotion-0 {
  position: relative;
  width: 100%;
  min-height: calc(100vh - 420px);
  padding-top: 16px;
  background-color: #FFFFFF;
  -webkit-filter: drop-shadow(0px 0px 8px rgba(131, 133, 136, 0.2));
  filter: drop-shadow(0px 0px 8px rgba(131, 133, 136, 0.2));
}

.emotion-0>div {
  max-width: 1024px;
  width: 100%;
  margin: 0 auto;
}

.emotion-0 .title {
  font-family: Hyundai Sans Head KR,sans-serif;
  font-size: 16px;
  font-weight: 500;
  line-height: 24px;
  letter-spacing: -0.03em;
  color: #212122;
  padding-bottom: 12px;
}

<div
  class="emotion-0"
  data-testid="main"
>
  <div />
</div>
`;
